k8s/nginxdemo-deployment.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: apps/v1beta1 # for versions after 1.8.0 use apps/v1beta2
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  selector:
+    matchLabels:
+      app: nginx
+  replicas: 2 # tells deployment to run 2 pods matching the template
+  template: # create pods using pod definition in this template
+    metadata:
+      # unlike pod-nginx.yaml, the name is not included in the meta data as a unique name is
+      # generated from the deployment name
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: bitnami/nginx
+        ports:
+        - containerPort: 80
```

Comment:
this is just an example/demo. the only difference from the example in the kubernetes docs is that it is using the bitnami nginx container image which does not require root, so that it works on OpenShift.